### ISSUE TITLE
Default memory limit in operator CSV is too low

### ISSUE THREAD

**Issue by bukovjanmic** [elastic/cloud-on-k8s#5177]
The default memory limit for Elastic operator (512Mi) is too low and as a consequence, elastic operator is OOMKilled when reconciling multiple Elastic resources.

```
apiVersion: operators.coreos.com/v1alpha1
kind: ClusterServiceVersion
metadata:
  annotations:
    certified: 'false'
...
spec:
...
                      - name: OPERATOR_IMAGE
                        value: >-
                          registry.connect.redhat.com/elastic/eck-operator@sha256:6555ff4bfd7823f2ab2557b4c44fb07e6227ca9f54a0c960d3a14329ee9462bc
                    image: >-
                      registry.connect.redhat.com/elastic/eck-operator@sha256:6555ff4bfd7823f2ab2557b4c44fb07e6227ca9f54a0c960d3a14329ee9462bc
                    name: manager
                    ports:
                      - containerPort: 9443
                        name: https-webhook
                        protocol: TCP
                    resources:
                      limits:
                        cpu: '1'
                        **memory: 2Gi**
                      requests:
                        cpu: 100m
                        memory: 150Mi
...
```

Increasing the memory limit to 1GI or 2Gi resolves the problem for us. We need to do this on each operator update via OLM, though.

This happened with 1.8.x and also with the new 1.9.0 version.

Please increase the memory limit to at least 1Gi to prevent this issue.

Thanks,

Michal Bukovjan

**Comment by pebrc** [comment]
@bukovjanmic thanks for reporting this. Could you share some numbers for context? For example how many Elasticsearch clusters or other Elastic resources you are running. But also, if you can, information about the overall size of your Kubernetes cluster (number of Pods, number of nodes) would be relevant. I suspect the Kubernetes go-client with its cache of all the objects we work with (so Pods, StatefulSets, Deployments, Nodes, Secrets, ConfigMaps etc)  to be responsible here, which increases the memory footprint of the operator in relation to the number of resources, but it would be helpful to verify that suspicion.

**Comment by pebrc** [comment]
We can definitely look into adjusting the resource limits for the operator. 

In the meantime, however, I wonder looking at your original post whether setting the higher memory limit could be done via the subscription instead of manipulating the ClusterServiceVersion directly: 

```
kind: Subscription
metadata:
  name: my-operator # just an example will be different in your installation
spec:
  package: elastic-cloud-eck # actual package name depends on your OLM flavour 
  channel: stable
  config:
    resources:
      requests:
        memory: "100Mi"
        cpu: "150m"
      limits:
        memory: "1Gi"
        cpu: "1"
```

This would have the advantage that the memory limits/requests would be kept at the modified values when you upgrade the operator IIUC.

**Comment by bukovjanmic** [comment]
Probably yes. I tried setting it in the Subscription object as you suggest, but so far no effect - in ClusterServiceVersion the original resource values remained.

However, the original default values in ClusterServiceVersion were not set by us either, so possibly if it is taken from Subscription object, that could be a better place.

The only action we took was increase the memory limit values in ClusterServiceVersion that was already there. By default, there is no resource configuration in Subscription object.

**Comment by pebrc** [comment]
> Probably yes. I tried setting it in the Subscription object as you suggest, but so far no effect - in ClusterServiceVersion the original resource values remained

The effect of the subscription config would not be observable on the CSV but on the operator Deployment and subsequently the Pods that are being created by OLM. I tried the example I posted on OCP on a subscription created by the OCP OperatorHub integration and the ECK operator Deployment was recreated with the updated memory limits. 

I think in general CSVs are not meant to be edited by the user as they are just copied out of the catalog and are ultimately created by us. If you want to adjust settings then the subscription config is the way to go. There is some documentation around this in the OLM repo: 

https://github.com/operator-framework/operator-lifecycle-manager/blob/master/doc/design/subscription-config.md

**Comment by bukovjanmic** [comment]
You are of course right. We just searched for a place to change the defaultts and CSV was the only place where we found the actual (low) values to change. We did not know about Subscription config (and when installing via OLM, we left everything at default).

So if it is configurable in Subscription, or better yet, if the defaults (for operator) are outright increased by default, that would be the best solution.
